File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/custom_tool.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.


from .tool import Tool
from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class CustomTool(Tool):
    """
    Derived model for Custom Tool
    """

    def __init__(self, **kwargs):
        """
        Initializes a new CustomTool object with values from keyword arguments. The default value of the :py:attr:`~oci.aidataplatform_dp.models.CustomTool.tool_type` attribute
        of this class is ``CUSTOM`` and it should not be changed.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param key:
            The value to assign to the key property of this CustomTool.
        :type key: str

        :param display_name:
            The value to assign to the display_name property of this CustomTool.
        :type display_name: str

        :param workspace_key:
            The value to assign to the workspace_key property of this CustomTool.
        :type workspace_key: str

        :param description:
            The value to assign to the description property of this CustomTool.
        :type description: str

        :param tool_type:
            The value to assign to the tool_type property of this CustomTool.
            Allowed values for this property are: "RAG", "SQL", "PROMPT", "NL2SQL", "MCP", "CUSTOM", "HTTP", "REFERENCE"
        :type tool_type: str

        :param is_reference:
            The value to assign to the is_reference property of this CustomTool.
        :type is_reference: bool

        :param position_x:
            The value to assign to the position_x property of this CustomTool.
        :type position_x: float

        :param position_y:
            The value to assign to the position_y property of this CustomTool.
        :type position_y: float

        :param properties:
            The value to assign to the properties property of this CustomTool.
        :type properties: dict(str, object)

        :param time_created:
            The value to assign to the time_created property of this CustomTool.
        :type time_created: datetime

        :param time_updated:
            The value to assign to the time_updated property of this CustomTool.
        :type time_updated: datetime

        :param created_by:
            The value to assign to the created_by property of this CustomTool.
        :type created_by: str

        :param updated_by:
            The value to assign to the updated_by property of this CustomTool.
        :type updated_by: str

        :param tool_provider:
            The value to assign to the tool_provider property of this CustomTool.
        :type tool_provider: str

        :param tool_type_name:
            The value to assign to the tool_type_name property of this CustomTool.
        :type tool_type_name: str

        :param input_schema:
            The value to assign to the input_schema property of this CustomTool.
        :type input_schema: dict(str, object)

        :param tool_config:
            The value to assign to the tool_config property of this CustomTool.
        :type tool_config: oci.aidataplatform_dp.models.CustomToolConfiguration

        """
        self.swagger_types = {
            'key': 'str',
            'display_name': 'str',
            'workspace_key': 'str',
            'description': 'str',
            'tool_type': 'str',
            'is_reference': 'bool',
            'position_x': 'float',
            'position_y': 'float',
            'properties': 'dict(str, object)',
            'time_created': 'datetime',
            'time_updated': 'datetime',
            'created_by': 'str',
            'updated_by': 'str',
            'tool_provider': 'str',
            'tool_type_name': 'str',
            'input_schema': 'dict(str, object)',
            'tool_config': 'CustomToolConfiguration'
        }

        self.attribute_map = {
            'key': 'key',
            'display_name': 'displayName',
            'workspace_key': 'workspaceKey',
            'description': 'description',
            'tool_type': 'toolType',
            'is_reference': 'isReference',
            'position_x': 'positionX',
            'position_y': 'positionY',
            'properties': 'properties',
            'time_created': 'timeCreated',
            'time_updated': 'timeUpdated',
            'created_by': 'createdBy',
            'updated_by': 'updatedBy',
            'tool_provider': 'toolProvider',
            'tool_type_name': 'toolTypeName',
            'input_schema': 'inputSchema',
            'tool_config': 'toolConfig'
        }

        self._key = None
        self._display_name = None
        self._workspace_key = None
        self._description = None
        self._tool_type = None
        self._is_reference = None
        self._position_x = None
        self._position_y = None
        self._properties = None
        self._time_created = None
        self._time_updated = None
        self._created_by = None
        self._updated_by = None
        self._tool_provider = None
        self._tool_type_name = None
        self._input_schema = None
        self._tool_config = None
        self._tool_type = 'CUSTOM'

    @property
    def tool_provider(self):
        """
        Gets the tool_provider of this CustomTool.
        The provider of the tool, default is AIDP


        :return: The tool_provider of this CustomTool.
        :rtype: str
        """
        return self._tool_provider

    @tool_provider.setter
    def tool_provider(self, tool_provider):
        """
        Sets the tool_provider of this CustomTool.
        The provider of the tool, default is AIDP


        :param tool_provider: The tool_provider of this CustomTool.
        :type: str
        """
        self._tool_provider = tool_provider

    @property
    def tool_type_name(self):
        """
        Gets the tool_type_name of this CustomTool.
        The type name for this tool


        :return: The tool_type_name of this CustomTool.
        :rtype: str
        """
        return self._tool_type_name

    @tool_type_name.setter
    def tool_type_name(self, tool_type_name):
        """
        Sets the tool_type_name of this CustomTool.
        The type name for this tool


        :param tool_type_name: The tool_type_name of this CustomTool.
        :type: str
        """
        self._tool_type_name = tool_type_name

    @property
    def input_schema(self):
        """
        Gets the input_schema of this CustomTool.
        The list of named properties in the inputSchema, along with the default value and description of each property


        :return: The input_schema of this CustomTool.
        :rtype: dict(str, object)
        """
        return self._input_schema

    @input_schema.setter
    def input_schema(self, input_schema):
        """
        Sets the input_schema of this CustomTool.
        The list of named properties in the inputSchema, along with the default value and description of each property


        :param input_schema: The input_schema of this CustomTool.
        :type: dict(str, object)
        """
        self._input_schema = input_schema

    @property
    def tool_config(self):
        """
        Gets the tool_config of this CustomTool.

        :return: The tool_config of this CustomTool.
        :rtype: oci.aidataplatform_dp.models.CustomToolConfiguration
        """
        return self._tool_config

    @tool_config.setter
    def tool_config(self, tool_config):
        """
        Sets the tool_config of this CustomTool.

        :param tool_config: The tool_config of this CustomTool.
        :type: oci.aidataplatform_dp.models.CustomToolConfiguration
        """
        self._tool_config = tool_config

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/CustomTool.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * Derived model for Custom Tool
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=CustomTool.Builder.class)
@com.fasterxml.jackson.annotation.JsonTypeInfo(use=com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, include=com.fasterxml.jackson.annotation.JsonTypeInfo.As.PROPERTY, property="toolType")

public final class CustomTool extends Tool {
    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
        @com.fasterxml.jackson.annotation.JsonProperty("key")
        private String key;

            public Builder key(String key) {
            this.key = key;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("displayName")
        private String displayName;

            public Builder displayName(String displayName) {
            this.displayName = displayName;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("workspaceKey")
        private String workspaceKey;

            public Builder workspaceKey(String workspaceKey) {
            this.workspaceKey = workspaceKey;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("description")
        private String description;

            public Builder description(String description) {
            this.description = description;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("isReference")
        private Boolean isReference;

            public Builder isReference(Boolean isReference) {
            this.isReference = isReference;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("positionX")
        private Float positionX;

            public Builder positionX(Float positionX) {
            this.positionX = positionX;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("positionY")
        private Float positionY;

            public Builder positionY(Float positionY) {
            this.positionY = positionY;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("properties")
        private java.util.Map<String, Object> properties;

            public Builder properties(java.util.Map<String, Object> properties) {
            this.properties = properties;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
        private java.util.Date timeCreated;

            public Builder timeCreated(java.util.Date timeCreated) {
            this.timeCreated = timeCreated;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
        private java.util.Date timeUpdated;

            public Builder timeUpdated(java.util.Date timeUpdated) {
            this.timeUpdated = timeUpdated;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("createdBy")
        private String createdBy;

            public Builder createdBy(String createdBy) {
            this.createdBy = createdBy;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("updatedBy")
        private String updatedBy;

            public Builder updatedBy(String updatedBy) {
            this.updatedBy = updatedBy;
            return this;
            }
            /**
     * The provider of the tool, default is AIDP
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("toolProvider")
        private String toolProvider;

                /**
         * The provider of the tool, default is AIDP
         * @param toolProvider the value to set
         * @return this builder
         **/
        

        public Builder toolProvider(String toolProvider) {
        this.toolProvider = toolProvider;
        return this;
        }
            /**
     * The type name for this tool
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("toolTypeName")
        private String toolTypeName;

                /**
         * The type name for this tool
         * @param toolTypeName the value to set
         * @return this builder
         **/
        

        public Builder toolTypeName(String toolTypeName) {
        this.toolTypeName = toolTypeName;
        return this;
        }
            /**
     * The list of named properties in the inputSchema, along with the default value and description of each property
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("inputSchema")
        private java.util.Map<String, Object> inputSchema;

                /**
         * The list of named properties in the inputSchema, along with the default value and description of each property
         * @param inputSchema the value to set
         * @return this builder
         **/
        

        public Builder inputSchema(java.util.Map<String, Object> inputSchema) {
        this.inputSchema = inputSchema;
        return this;
        }
        
        @com.fasterxml.jackson.annotation.JsonProperty("toolConfig")
        private CustomToolConfiguration toolConfig;

        

        public Builder toolConfig(CustomToolConfiguration toolConfig) {
        this.toolConfig = toolConfig;
        return this;
        }


        public CustomTool build() {
            CustomTool model = new CustomTool(this.key
                    , this.displayName
                    , this.workspaceKey
                    , this.description
                    , this.isReference
                    , this.positionX
                    , this.positionY
                    , this.properties
                    , this.timeCreated
                    , this.timeUpdated
                    , this.createdBy
                    , this.updatedBy
                    , this.toolProvider
                    , this.toolTypeName
                    , this.inputSchema
                    , this.toolConfig);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(CustomTool model) {
                this.key(model.getKey());
    this.displayName(model.getDisplayName());
    this.workspaceKey(model.getWorkspaceKey());
    this.description(model.getDescription());
    this.isReference(model.getIsReference());
    this.positionX(model.getPositionX());
    this.positionY(model.getPositionY());
    this.properties(model.getProperties());
    this.timeCreated(model.getTimeCreated());
    this.timeUpdated(model.getTimeUpdated());
    this.createdBy(model.getCreatedBy());
    this.updatedBy(model.getUpdatedBy());
    this.toolProvider(model.getToolProvider());
    this.toolTypeName(model.getToolTypeName());
    this.inputSchema(model.getInputSchema());
    this.toolConfig(model.getToolConfig());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    
    @Deprecated
    public CustomTool(String key, String displayName, String workspaceKey, String description, Boolean isReference, Float positionX, Float positionY, java.util.Map<String, Object> properties, java.util.Date timeCreated, java.util.Date timeUpdated, String createdBy, String updatedBy, String toolProvider, String toolTypeName, java.util.Map<String, Object> inputSchema, CustomToolConfiguration toolConfig) {
        super(key, displayName, workspaceKey, description, isReference, positionX, positionY, properties, timeCreated, timeUpdated, createdBy, updatedBy);
        this.toolProvider = toolProvider;
        this.toolTypeName = toolTypeName;
        this.inputSchema = inputSchema;
        this.toolConfig = toolConfig;
    }


        /**
     * The provider of the tool, default is AIDP
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("toolProvider")
    private final String toolProvider;

        /**
     * The provider of the tool, default is AIDP
     * @return the value
     **/
    
    public String getToolProvider() {
        return toolProvider;
    }


        /**
     * The type name for this tool
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("toolTypeName")
    private final String toolTypeName;

        /**
     * The type name for this tool
     * @return the value
     **/
    
    public String getToolTypeName() {
        return toolTypeName;
    }


        /**
     * The list of named properties in the inputSchema, along with the default value and description of each property
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("inputSchema")
    private final java.util.Map<String, Object> inputSchema;

        /**
     * The list of named properties in the inputSchema, along with the default value and description of each property
     * @return the value
     **/
    
    public java.util.Map<String, Object> getInputSchema() {
        return inputSchema;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("toolConfig")
    private final CustomToolConfiguration toolConfig;

    
    public CustomToolConfiguration getToolConfig() {
        return toolConfig;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("CustomTool(");
        sb.append("super=").append(super.toString(includeByteArrayContents));
        sb.append(", toolProvider=").append(String.valueOf(this.toolProvider));
        sb.append(", toolTypeName=").append(String.valueOf(this.toolTypeName));
        sb.append(", inputSchema=").append(String.valueOf(this.inputSchema));
        sb.append(", toolConfig=").append(String.valueOf(this.toolConfig));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof CustomTool)) {
            return false;
        }

        CustomTool other = (CustomTool) o;
        return java.util.Objects.equals(this.toolProvider, other.toolProvider) &&
            java.util.Objects.equals(this.toolTypeName, other.toolTypeName) &&
            java.util.Objects.equals(this.inputSchema, other.inputSchema) &&
            java.util.Objects.equals(this.toolConfig, other.toolConfig) &&
            super.equals(other);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = super.hashCode();
        result = (result * PRIME) + (this.toolProvider == null ? 43 : this.toolProvider.hashCode());
        result = (result * PRIME) + (this.toolTypeName == null ? 43 : this.toolTypeName.hashCode());
        result = (result * PRIME) + (this.inputSchema == null ? 43 : this.inputSchema.hashCode());
        result = (result * PRIME) + (this.toolConfig == null ? 43 : this.toolConfig.hashCode());
        return result;
    }


}
